pkg/arvo/neo/cod/std/src/fil/types.txt

```diff
@@ -0,0 +1,460 @@
+
+`bowl:neo`, similar to `bowl:gall`.
+
+```
++$  bowl
+  $:  src=name
+      our=@p
+      were=pith
+      here=pith
+      now=@da
+      eny=@uvJ
+      deps=(map term (pair pith lore))
+      kids=lore
+  ==
+```
+
+- `src`: source ship and path [of what?].
+- `our`: our ship.
+- `were`: Deprecated; use `here` instead.
+- `here`: location of this shrub.
+- `now`: current datetime.
+- `eny`: entropy.
+- `deps`: ???
+- `kids`: ???
+
+[above are dependencies and descendants yeah but need `lore` to explain the types and what they do here]
+
+## `card`
+
+An instruction for a shrub, similar to a `card:agent:gall`.
+
+```
++$  card  (pair pith note)
+```
+
+The `pith` is the shrub this card will be sent to. The `note` will be of type `%make`, `%poke`, `%tomb`, or `%cull`.
+
+## `care`
+
+Perspective on a path.
+
+```
++$  care
+  $~  %x
+  $?  %x
+      %y
+      %z
+      %a
+      %b
+      %c
+  ==
+```
+
+The cares `%x`, `%y`, and `%z` appear throughout the system: you’ll use them while defining `kids` and `deps`, and state updates will be head-tagged with one of these cares.
+
+[not sure about %a, %b, %c; i know the correspond to x y z but not sure about usage]
+
+## `conf`
+
+Dependency configuration for a new shrub.
+
+```
++$  conf  (map term pith)
+```
+
+This can be included in a `%make` card when we want to pass dependencies into a new shrub.
+
+The `term`s are arbitrary tags the new shrub has defined, and the `pith`s are locations of the relevant shrubs whose state will be tracked for the new shrub to respond to.
+
+## `curb`
+
+Constraint on the state of a shrub.
+
+```
++$  curb
+  $~  [%pro %$]
+  $%  [%or p=(list curb)]
+      [%only p=stud]
+      [%rol p=stud q=curb]
+      [%not p=curb q=curb]
+      [%pro p=stud]
+      [%any ~]
+  ==
+```
+
+[explain each individually]
+
+[how can they be combined, why would you do that?]
+
+## `dare`
+
+A `%y` or `%z` care. (Mnemonic: "dare is a downward care".)
+
+```
++$  dare  ?(%y %z)
+```
+
+Used while constraining descendants.
+
+## `deed`
+
+Whether or not a depdenency must already exist in order for this shrub to build.
+
+```
++$  deed
+  $@  ?
+  $:  time=(unit @dr)
+      req=?
+  ==
+```
+
+[in what case is `time` used?]
+
+## `deps`
+
+Dependency map.
+
+```
++$  deps  band
+```
+
+## `dita`
+
+A cell of either `%.y` and an `iota` or `%.n` and an `aura`.
+
+```
++$  dita  (each iota aura)
+```
+
+See `pish` for usage.
+
+## `ever`
+
+Total version numbers for a shrub and its descendants.
+
+```
++$  ever
+  $:  exe=lock
+      why=lock
+      zed=lock
+    ::
+      shrub-life=@ud
+      ship-life=@ud
+      ship-rift=@ud
+    ::
+      =time
+  ==
+```
+
+- “Exe” is only incremented when the shrub itself updates it state. (i.e. `%x` care.)
+- “Why” is incremented when the shrub or any of its immediate children updates. (i.e. `%y` care.)
+- “Zed” is incremented when the shrub or any of its descendants updates. (i.e. `%z` care.)
+
+[what about shrub-life, ship-rift, etc?]
```

Comment:
most of these are implementation detail. exe, why and zed are important.
each of the fields that correspond to a care have two version numbers associated with them. One of which is the 'data' version, which is what you think of when you think of a 'case'. For exe, It is the number of times that the data has changed. For why, it's the sum of exe and the children's exe. For zed, it's the sum of exe and all descendants' exe. 

The other is the 'shape' version, which is the number of times the shape of this response has changed. For exe, this is just the number of times that it has transitioned from full to empty or vice-versa. 
e.g.
```hoon
%make -> %cull -> %make  :: exe shape of 3
%make :: exe shape of 1
(never bound) :: exe shape of 0
```
the shape versions for why and zed follow the same summation logic as the data versions, except being calculated off the shape exe. The exe shape version is actually also the breach count for that node (i.e. the shrub's rift). 

All of the other fields are needed to maintain referential transparency, however they're unimportant to document rn